src/main/java/roomescape/business/service/ReservationService.java

```diff
@@ -38,26 +40,68 @@ public class ReservationService {
 
-        if (reservationStatus == ReservationStatus.RESERVED && reservationRepository.isDuplicateDateAndTimeAndTheme(date,
-                reservationTime.startTimeValue(), theme.getId())) {
+    public ReservationDto addAndGet(final LocalDate date, final String timeIdValue,
+                                    final String themeIdValue,
+                                    final String userIdValue, final ReservationStatus reservationStatus,
+                                    final String paymentKey, final String orderId, final Long amount) {
+        User user = getUser(userIdValue);
+        ReservationTime reservationTime = getReservationTime(timeIdValue);
+        Theme theme = getTheme(themeIdValue);
+
+        if (reservationStatus == ReservationStatus.RESERVED &&
+                reservationRepository.isDuplicateDateAndTimeAndTheme(date, reservationTime.startTimeValue(),
+                        theme.getId())) {
             throw new DuplicatedException(RESERVATION_DUPLICATED);
         }
+        Reservation reservation = Reservation.create(user, date, reservationTime, theme, reservationStatus,
+                LocalDateTime.now());
+        if (reservationStatus == ReservationStatus.RESERVED) {
+            paymentClient.approvePayment(new PaymentApproveDto(paymentKey, orderId, amount));
+        }
+        reservationRepository.save(reservation);
+        waitingService.updateWaitingReservations(reservation);
+        return ReservationDto.fromEntity(reservation);
+    }
+
+    public ReservationDto addAndGetWithoutPayment(final LocalDate date, final String timeIdValue,
+                                                  final String themeIdValue,
+                                                  final String userIdValue, final ReservationStatus reservationStatus) {
+        User user = getUser(userIdValue);
+        ReservationTime reservationTime = getReservationTime(timeIdValue);
+        Theme theme = getTheme(themeIdValue);
 
-        Reservation reservation = Reservation.create(user, date, reservationTime, theme, reservationStatus, LocalDateTime.now());
+        if (reservationStatus == ReservationStatus.RESERVED && reservationRepository.isDuplicateDateAndTimeAndTheme(
+                date,
+                reservationTime.startTimeValue(), theme.getId())) {
+            throw new DuplicatedException(RESERVATION_DUPLICATED);
+        }
+        Reservation reservation = Reservation.create(user, date, reservationTime, theme, reservationStatus,
+                LocalDateTime.now());
         reservationRepository.save(reservation);
         waitingService.updateWaitingReservations(reservation);
         return ReservationDto.fromEntity(reservation);
     }
 
+    private Theme getTheme(String themeIdValue) {
+        Theme theme = themeRepository.findById(Id.create(themeIdValue))
+                .orElseThrow(() -> new NotFoundException(THEME_NOT_EXIST));
+        return theme;
```

Comment:
```suggestion
        return themeRepository.findById(Id.create(themeIdValue))
                .orElseThrow(() -> new NotFoundException(THEME_NOT_EXIST));
```
변수로 뺐을 때 변수로써 활용되지 않는다면 return문이 하나 더 늘어나므로 읽어야하는 줄이 1줄 늘어난다라고도 볼 수 있겠네요.

Reply:
불필요한 코드가 있었네요 😂 
리팩토링 과정에서 해당 메서드는 제거되었습니다..!